pandas/io/formats/format.py

```diff
@@ -1514,18 +1514,25 @@ def __init__(
 
     def _format_strings(self) -> List[str]:
         """ we by definition have DO NOT have a TZ """
-        values = self.values
-
-        if not isinstance(values, DatetimeIndex):
-            values = DatetimeIndex(values)
+        values = np.asarray(self.values)
+        flat_values = values.ravel() if len(values.shape) > 1 else values
+        flat_values = DatetimeArray(flat_values)
 
         if self.formatter is not None and callable(self.formatter):
-            return [self.formatter(x) for x in values]
+            fmt_values = [self.formatter(x) for x in flat_values]
+        else:
+            fmt_values = flat_values._format_native_types(
+                na_rep=self.nat_rep, date_format=self.date_format
+            )
 
-        fmt_values = values._data._format_native_types(
-            na_rep=self.nat_rep, date_format=self.date_format
-        )
-        return fmt_values.tolist()
+        if len(values.shape) > 1:
```

Comment:
`if values.ndim > 1:`

Reply:
I wasn't sure if all possible `value` classes have that property, but they should if they also have shape. Let me change that and rerun tests.